### ISSUE TITLE
Double click with right button in terminal window causes Xorg to crash

### ISSUE THREAD

**Issue by quatrecouleurs** [mate-desktop/mate-terminal#171]
In archlinux 64, using radeon driver with AMD-A10 APU, I have the following bug:
- Open mate-terminal
- Double-click with the right button

-> The context menu just shows an empty window
-> Xorg crashes strangely ? (Freezes, no interaction possible in the windows, panel, etc)

`sudo killall marco` in TTY2 works and changes the screen output, as does `sudo killall mate-panel`. I can restart those two, but still I cannot interact either with the mouse or the keyboard.

I tried to replicate this in gnome-terminal, no problem so far.

I didn't reset all custom setting though but will test it later. I just found it quite critical so I reported it ASAP. Thanks !

**Comment by honza889** [comment]
Confirm - I am able to reproduce with mate-terminal **1.16.2** on Fedora 25.

**Comment by GUiHKX** [comment]
Just experienced this same issue on Arch Linux. However, Xorg doesn't crash, but the entire desktop freezes, so you have to log into a different tty and kill mate-terminal, then everything gets back to normal.

**Comment by raveit65** [comment]
I can reproduce this with fedora 25 VM.
```
[rave@f25 ~]$ rpm -qa mate-terminal
mate-terminal-1.16.2-1.fc25.x86_64
```
But not any more with f26 on bare metal
```
[rave@mother ~]$ rpm -qa mate-terminal
mate-terminal-1.18.0-1.fc26.x86_64
```
You better post you version here when you post a issue at mate github.
I do not have any interest to find out for myself which version archlinux use ;-)
Maybe you think archlinux is the only linux distro :-)

**Comment by GUiHKX** [comment]
Arch uses the very latest version, **1.18.0**. Arch is so good that sometimes I forget that other distros exist. ;)

**Comment by raveit65** [comment]
Thank you.
Does Arch use libinput X driver with mate-1.18?
For F26 i switched MATE to libinput X driver, maybe this is the reason for different results with m-t-1.18.

I can't reproduce the issue any more with 
[rave@f25 ~]$ rpm -qa mate-terminal
mate-terminal-1.16.1-1.fc25.x86_64
in f25 VM.
@honza889 
Can you please downgrade for confirming?
sudo dnf downgrade mate-terminal

**Comment by GUiHKX** [comment]
No problem! I don't have too much knowledge about the technical stuff, but I'll try to do my best to help. 
 I have installed **xf86-input-libinput 0.25.0-1**, and **xf86-input-vmmouse 13.1.0-3** (I'm running Arch on a VM).

**Comment by willysr** [comment]
confirmed on Slackware-Current. Simply right click 3x and it will freeze the desktop
MATE 1.18.0
XOrg 1.19.3
xf86-input-libinput 0.25.0
mesa 17.0.2
xf86-input-vmmouse-13.1.0

**Comment by sc0w** [comment]
I can't reproduce it with mate-terminal 1.16.1 and I can reproduce the issue with mate-terminal 1.18.0

**Comment by sc0w** [comment]
gdb show me this warning several times on the freezing:

`(mate-terminal:4620): Gtk-CRITICAL **: _gtk_widget_captured_event: assertion 'WIDGET_REALIZED_FOR_EVENT (widget, event)' failed`

**Comment by raveit65** [comment]
looks like culprit commit is https://github.com/mate-desktop/mate-terminal/commit/575fc201f08cefecf469344ba3b2b02f1174fbc1

**Comment by monsta** [comment]
It gets worse. At first I couldn't reproduce it at all. Some weeks later and some updates later, I could kill all mouse interaction in the session with it. Keyboard still worked, so I could log out normally and then restart the session. Some more days and updates passed by, and now I can fully reproduce this, not every time though.

So far, I only tried it on Debian Testing with GTK+ 3.22.11.

I have no idea how we could possibly cause this. We're too high-level to cause such issues for X server, for us it's behind GTK+, GDK and whatever. Sounds like another GTK+3 regression to me, especially because there's an assertion in some *internal* GTK+ function (judging by underscore prefix in ```_gtk_widget_captured_event``` name).

**Comment by monsta** [comment]
Looks like you can do it in gnome-terminal as well, just with different steps to reproduce:

https://bugzilla.redhat.com/1435911

**Comment by sc0w** [comment]
@monsta you tested my PR? https://github.com/mate-desktop/mate-terminal/pull/173

**Comment by GUiHKX** [comment]
@sc0w's patch worked for me!

```
Arch Linux x64
mate-terminal 1.18.0
```

At least isn't freezing like before when I double right-click on the terminal window.

Thanks a lot!

**Comment by monsta** [comment]
I didn't try #173 yet, will check it

**Comment by monsta** [comment]
Ok, for me it's fixed, thanks @sc0w :slightly_smiling_face: 
Runtime warnings are gone as well.

I still think there's some internal issue in GTK+ though.

**Comment by monsta** [comment]
Also tested it in Ubuntu 16.10 with mate-terminal 1.16.2 and GTK+ 3.20.9. The bug is reproducible there, and the fix works too.

**Comment by monsta** [comment]
Next system, Mint 18.1 (based on Ubuntu 16.04), with mate-terminal 1.16.2 and GTK+ 3.18.9. Same here: the bug is reproducible and the fix works.

**Comment by monsta** [comment]
And one more, LMDE 2 (based on Debian Jessie), with mate-terminal 1.18.0 and GTK+ 3.14.5. Surprise, it's reproducible even with this old GTK+ version... and the fix works once again.

**Comment by monsta** [comment]
BTW, did anyone notice that the clock applet still keeps working (time doesn't stop)? So the bug only affects keyboard and mouse input, it doesn't freeze the whole X server.

**Comment by raveit65** [comment]
> BTW, did anyone notice that the clock applet still keeps working (time doesn't stop)? So the bug only affects keyboard and mouse input, it doesn't freeze the whole X server.

Yes , i noticed that too.
Bug confirmed with 1.16.2 and 1.18.0 and PR fixes the issue, and i see no regression in gnome-wayland session.

**Comment by monsta** [comment]
The fix is pushed to 1.18 and 1.16 branches as well.
